### ISSUE TITLE
Implement borsh serialization/deserialization for the core types in celestia-types

### ISSUE THREAD

**Issue by zvolin** [eigerco/celestia-node-rs#155]
Borsh is a popular and more efficient alternative to serde, we could support it behind a feature flag

**Comment by Wondertan** [comment]
Would you use it for storage?

**Comment by zvolin** [comment]
no, I think protobuf is anyway better there as it's not rust-specific. This is just a general feature that some people may want to use with the types we provide. I know eg. sovereign is using borsh
